spring-app/src/main/java/pl/edu/agh/cqm/service/SchedulerService.java

```diff
@@ -1,27 +1,34 @@
 package pl.edu.agh.cqm.service;
 
-import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
+import pl.edu.agh.cqm.configuration.CqmConfiguration;
+import pl.edu.agh.cqm.data.repository.ConfigSampleRepository;
 
 @Service
 @Configuration
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class SchedulerService {
     private final PingService pingService;
     private final ThroughputService throughputService;
+    private final CqmConfiguration cqmConfiguration;
+    private final ConfigSampleRepository configSampleRepository;
 
+    private int schedulePingCounter = 0;
 
     @Scheduled(fixedDelay = 1)
     public void scheduledThroughput(){
         throughputService.doMeasurement();
     }
 
-
-    @Scheduled(fixedRateString = "#{cqmConfiguration.getActiveSamplingRate() * 60 * 1000}")
-    public void scheduledPing(){
-        pingService.doMeasurement();
+    @Scheduled(fixedRate = 60000)
+    public void schedulePing() {
+        if (schedulePingCounter == 0) {
+            pingService.doMeasurement();
+            schedulePingCounter = configSampleRepository.findFirstByOrderByIdDesc().getActiveSamplingRate();
```

Comment:
it might not be guaranteed, that the newest id is also the largest. Better sort by timestamp

Reply:
configuration should be accessed through a more readable method from `ParameterService`